state/factory.go

```diff
@@ -271,24 +271,29 @@ func (sf *factory) CommitStateChanges(blockHeight uint64, tsf []*action.Transfer
 	if blockHeight > 0 && len(sf.cachedCandidates) == 0 {
 		candidates, err := sf.getCandidates(blockHeight - 1)
 		if err != nil {
+			logger.Error().Err(err).Msgf("Failed to get previous candidates on height %d", blockHeight-1)
```

Comment:
No need to log the error separately. Because you have already captured and logged error above.